api/xverse.ts

```diff
@@ -146,16 +149,34 @@ export async function getBinaceSignature(srcData: string): Promise<SignedUrlResp
     });
 }
 
-export async function sponsorTransaction(signedTx: StacksTransaction): Promise<string> {
+export async function sponsorTransaction(signedTx: StacksTransaction): Promise<string | null> {
   const sponsorUrl = `${XVERSE_SPONSOR_URL}/v1/sponsor`;
 
   const data = {
     tx: signedTx.serialize().toString('hex'),
   };
 
-  return axios.post(sponsorUrl, data, { timeout: 45000 }).then((response) => {
-    return response.data.txid;
-  });
+  return axios
+    .post(sponsorUrl, data, { timeout: 45000 })
+    .then((response: AxiosResponse<SponsorTransactionResponse>) => {
+      return response.data.txid;
+    })
+    .catch(() => {
+      return null;
```

Comment:
we should pass the error along to the apps from here to handle on the client side